analysis/study_definition.py

```diff
@@ -40,70 +198,112 @@
+        include_date_of_match=True,
         include_month=True,
         return_expectations={
-            "incidence": 0.6,
-            "float": {"distribution": "normal", "mean": 35, "stddev": 10},
+            "date": {"latest": "2020-02-29"},
+            "float": {"distribution": "normal", "mean": 5, "stddev": 2},
+            "incidence": 0.95,
         },
     ),
-    # https://github.com/ebmdatalab/tpp-sql-notebook/issues/35
-    bp_sys=patients.mean_recorded_value(
-        systolic_blood_pressure_codes,
-        on_most_recent_day_of_measurement=True,
-        on_or_before="2020-02-01",
-        include_measurement_date=True,
+    hypertension=first_diagnosis_in_period(hypertension_codes),
+    chronic_respiratory_disease=first_diagnosis_in_period(chronic_respiratory_disease_codes),
+    copd=first_diagnosis_in_period(copd_codes),
+    chronic_liver_disease=first_diagnosis_in_period(chronic_liver_disease_codes),
+    stroke=first_diagnosis_in_period(stroke_codes),
+    lung_cancer=first_diagnosis_in_period(lung_cancer_codes),
+    haem_cancer=first_diagnosis_in_period(haem_cancer_codes),
+    other_cancer=first_diagnosis_in_period(other_cancer_codes),
+    #CKD
+    creatinine=patients.with_these_clinical_events(
+        creatinine_codes,
+        find_last_match_in_period=True,
+        between=["2018-12-01", "2020-02-29"],
+        returning="numeric_value",
+        include_date_of_match=True,
         include_month=True,
         return_expectations={
-            "incidence": 0.6,
-            "float": {"distribution": "normal", "mean": 80, "stddev": 10},
+            "float": {"distribution": "normal", "mean": 150.0, "stddev": 200.0},
+            "date": {"earliest": "2018-12-01", "latest": "2020-02-29"},
+            "incidence": 0.95,
         },
     ),
-    bp_dias=patients.mean_recorded_value(
-        diastolic_blood_pressure_codes,
-        on_most_recent_day_of_measurement=True,
-        on_or_before="2020-02-01",
+    #### end stage renal disease codes incl. dialysis / transplant 
+    esrf=patients.with_these_clinical_events(
+        ckd_codes,
+        on_or_before="2020-02-29",
+        return_last_date_in_period=True,
+        include_month=True,
+        return_expectations={"date": {"latest": "2020-02-29"}},
+    ),
+    ckd=first_diagnosis_in_period(ckd_codes),
+    organ_transplant=first_diagnosis_in_period(organ_transplant_codes),
+    # https://github.com/ebmdatalab/tpp-sql-notebook/issues/10
+    bmi=patients.most_recent_bmi(
+        on_or_after="2010-02-01",
+        minimum_age_at_measurement=16,
         include_measurement_date=True,
         include_month=True,
         return_expectations={
             "incidence": 0.6,
-            "float": {"distribution": "normal", "mean": 120, "stddev": 10},
+            "float": {"distribution": "normal", "mean": 35, "stddev": 10},
         },
     ),
     # https://github.com/ebmdatalab/tpp-sql-notebook/issues/54
     stp=patients.registered_practice_as_of(
```

Comment:
stp and msoa re not currently in the protocol.........

Reply:
We will add STP to the protocol and remove MSOA from the script.